core/include/cubos/core/geom/intersections.hpp

```diff
@@ -33,4 +34,24 @@ namespace cubos::core::geom
     CUBOS_CORE_API bool intersects(const Box& box1, const glm::mat4& localToWorld1, const Box& box2,
                                    const glm::mat4& localToWorld2, Intersection& intersect);
 
+    /// @brief Performs the Sutherland-Hodgman Clipping algorithm.
+    /// @param inputPolygon The polygon to perform the clipping on.
+    /// @param numClipPlanes Number of cliping planes.
+    /// @param clipPlanes Clipping planes
+    /// @param removeNotClipToPlane Whether to remove the points if they're outside the plane.
+    /// @return The polygon resulting from the clipping.
+    CUBOS_CORE_API std::list<glm::vec3> sutherlandHodgmanClipping(const std::list<glm::vec3>& inputPolygon,
```

Comment:
Any reason for an `std::list` btw? I think a `std::vector` would probably perform better. (Same for all other uses of it in this PR)

Reply:
I initially just followed along the example code, and I remember thinking about it too, but I don't remember if I saw anything or I just forgot to change. I will review and change if I see it's ok.